src/escpos/escpos.py

```diff
@@ -128,6 +128,27 @@ def __del__(self):
         """Call self.close upon deletion."""
         self.close()
 
+    @property
+    def device(self):
+        """Implements a self-open mechanism.
+
+        An attempt to get the property before open the connection
+        will cause the connection to open.
+        """
+        if self._device is False:
```

Comment:
Well, `bool(None)` returns `False` but `None` and `False` are different objects. Then `bool(None) == False`, `bool(None) is False` but `None != False` and `None is not False` because python checks if both are the same object, so yes, this way one can differentiate between the two states (objects).

Changing the state of the `device` property from 'not initialized' (False) to 'initialized but not opened' (None) was the way I found to auto-open the connection only on the first call, otherwise, as you guess, every check if the device exists would try to reopen the connection.
A side effect is that when the user instantiates a printer for the first time and then checks the value of `device` then the connection tries to be opened:
```
import escpos.printer as printer

>>> p = printer.Serial()
>>> print(p.device)
Serial printer Enabled
Serial<id=0x7f8342d78790, open=True>(port='/dev/ttyUSB0', baudrate=9600, bytesize=8, parity='N', stopbits=1, timeout=1, xonxoff=False, rtscts=False, dsrdtr=True)
```
but this behavior could be useful to check the connection before sending any commands.

Reply:
I agree. Is there then a way to restore the connection? (Apart from explicitly calling open()?)

Reply:
After commit 7103af083c2628006d6d1b53908f6dcfcc2b433d a closed connection will be reopened in the same way as the first time (self-open or explicit call to `open()`), which allows the reuse and|or reconfiguration of the object.

Reply:
Great, I think this is resolved then